tools/yamcs/proves_adapter.py

```
@@ -72,36 +73,70 @@ def _forward_tm_serial(
-    last_vc_count = -1  # not yet known
+    frames_sent_by_scid: Counter[int] = Counter()
+    last_vc_count: dict[int, int] = {}
     vc_frame_gaps = 0
     junk_bytes = 0
     stats_time = time.monotonic()
 
-    print(f"[TM] Scanning for frames (sync header={sync_header.hex(' ')})...")
+    sync_desc = " / ".join(f"{h.hex(' ')} (SCID {s})" for s, h in syncs)
+    print(f"[TM] Scanning for frames (sync headers: {sync_desc})...")
+
+    def _maybe_print_stats():
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_maybe_print_stats`

Add return type annotation: `None`

(ANN202)